libs/api/domains/university-of-iceland/src/lib/universityOfIceland.resolver.ts

```diff
@@ -0,0 +1,73 @@
+import { Args, Query, Resolver } from '@nestjs/graphql'
+
+import { Inject, UseGuards } from '@nestjs/common'
+import format from 'date-fns/format'
+import is from 'date-fns/locale/is'
+
+import {
+  CurrentUser,
+  IdsUserGuard,
+  Scopes,
+  ScopesGuard,
+} from '@island.is/auth-nest-tools'
+import type { User } from '@island.is/auth-nest-tools'
+import {
+  UniversityOfIcelandService,
+  NemandiGetLocaleEnum,
+  NemandiFerillFerillGetLocaleEnum,
+  StudentTrackTranscripts,
+} from '@island.is/clients/university-of-iceland'
+import { ApiScope } from '@island.is/auth/scopes'
+import { StudentInfoModel } from './models/studentInfo.model'
+import { StudentInfoDetailModel } from './models/studentInfoDetail.model'
+import { GetStudentInfoDetailInput } from './dto/getStudentInfoDetail.input'
+import { GetStudentInfoInput } from './dto/getStudentInfo.input'
+import { DownloadServiceConfig } from '@island.is/nest/config'
+import type { ConfigType } from '@island.is/nest/config'
+
+@UseGuards(IdsUserGuard, ScopesGuard)
+@Scopes(ApiScope.internal)
+@Resolver()
+export class UniversityOfIcelandResolver {
+  constructor(
+    private universityOfIcelandApi: UniversityOfIcelandService,
+    @Inject(DownloadServiceConfig.KEY)
+    private readonly downloadServiceConfig: ConfigType<
+      typeof DownloadServiceConfig
+    >,
+  ) {}
+
+  @Query(() => StudentInfoModel)
```

Comment:
Missing [namespace and correct naming convention](https://docs.devland.is/technical-overview/api-design-guide/graphql-naming-conventionsl) for all @InputType, @ObjectType and @Query.

E.g. `universityOfIcelandStudentInfo`.

Reply:
modified, better now?

Reply:
Not quite, here are examples:
```
// To name the query in the GQL schema
@Query(() => StudentInfoModel, { name: 'universityOfIcelandStudentInfo'})
async studentInfo(...) {...}
```

```
// To name model in GQL schema
@ObjectType('UniversityOfIcelandInstitution')
export class Institution {
  @Field(() => String)
  id!: string

  @Field(() => String)
  displayName!: string
}
```

```
// To name input types
@InputType('UniversityOfIcelandStudentInfoInput')
export class StudentInfoInput {
  @Field({ nullable: true })
  @IsNumber()
  @IsOptional()
  trackNumber?: number

  @Field()
  @IsString()
  locale!: string
}
```

The naming convention is to name queries, mutations, objects and input types with the domain name as prefix and then the operation or model name. But the names can be simplified in the code but the decorators used to control the naming for the GQL schema.

Reply:
ah ok, i've misunderstood the conventions, will update.